adafruit_tmp006.py

```diff
@@ -0,0 +1,182 @@
+"""
+
+import time
+import struct
+from micropython import const
+from adafruit_bus_device.i2c_device import I2CDevice
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_TMP006.git"
+
+# Default device I2C address.
+_TMP006_I2CADDR = const(0x40)
+
+# Register addresses.
+_TMP006_VOBJ = const(0x00)
+_TMP006_TAMB = const(0x01)
+_TMP006_CONFIG = const(0x02)
+_TMP006_MANUID = const(0xFE)
+_TMP006_DEVID = const(0xFF)
+
+# Config register values.
+_TMP006_CFG_RESET = const(0x8000)
+_TMP006_CFG_MODEON = const(0x7000)
+CFG_1SAMPLE = const(0x0000)
+CFG_2SAMPLE = const(0x0200)
+CFG_4SAMPLE = const(0x0400)
+CFG_8SAMPLE = const(0x0600)
+CFG_16SAMPLE = const(0x0800)
+_TMP006_CFG_DRDYEN = const(0x0100)
+_TMP006_CFG_DRDY = const(0x0080)
+
+class TMP006:
+    """Class to represent an Adafruit TMP006 non-contact temperature measurement
+    board.
+    """
+
+    # Class-level buffer for reading and writing data with the sensor.
+    # This reduces memory allocations but means the code is not re-entrant or
+    # thread safe!
+    _BUFFER = bytearray(4)
+
+    def __init__(self, i2c, address=_TMP006_I2CADDR, samplerate=CFG_16SAMPLE):
+        self._device = I2CDevice(i2c, address)
+        self._write_u16(_TMP006_CONFIG, _TMP006_CFG_RESET)
+        time.sleep(.5)
+        if samplerate not in (CFG_1SAMPLE, CFG_2SAMPLE, CFG_4SAMPLE, CFG_8SAMPLE,
+                              CFG_16SAMPLE):
+            raise ValueError('Unexpected samplerate value! Must be one of: ' \
+                'CFG_1SAMPLE, CFG_2SAMPLE, CFG_4SAMPLE, CFG_8SAMPLE, or CFG_16SAMPLE')
+        # Set configuration register to turn on chip, enable data ready output,
+        # and start sampling at the specified rate.
+        config = _TMP006_CFG_MODEON | _TMP006_CFG_DRDYEN | samplerate
+        self._write_u16(_TMP006_CONFIG, config)
+        # Check device ID match expected value.
+        dev_id = self.read_register(_TMP006_DEVID)
+        if dev_id != 0x67:
+            raise RuntimeError('Init failed - Did not find TMP006')
+
+    @property
+    def active(self):
+        """True if sensor is active."""
+        return self._read_u16(_TMP006_CONFIG) & _TMP006_CFG_MODEON != 0
+
+    @active.setter
+    def active(self, val):
+        if val:
+            self._wake()
+        else:
+            self._sleep()
+
+    @property
+    def temperature(self):
+        # pylint: disable=bad-whitespace, invalid-name, too-many-locals
+        """Read object temperature from TMP006 sensor."""
+        if not self.active:
+            raise RuntimeError('Can not read from sensor when inactive.')
+        while not self._data_ready():
+            pass
+        vobj = self.read_register(_TMP006_VOBJ)
+        vobj = bytearray((vobj & 0xFF, vobj >> 8 & 0xFF))
+        vobj = struct.unpack("<h", vobj)[0]
+        vobj = vobj * 156.25e-9  # volts
+        tamb = self.read_register(_TMP006_TAMB)
```

Comment:
split this into its own function like we do here
https://github.com/adafruit/Adafruit_Python_TMP/blob/master/Adafruit_TMP/TMP006.py#L135

Reply:
done